Yoast/wordpress-seo
# Optimize migrations framework

## Summary


This PR can be summarized in the following changelog entry:

* Ensures placing a file in the migrations directory with the right naming doesn't automatically load and execute it.

## Relevant technical choices:

* The Ruckusing dependency has been removed and replaced by our own implementation in the lib directory that's based on it.

## Test instructions

This PR can be tested by following these steps:

* Run composer install to ensure the ruckusing dependency is removed.
* Remove all the Yoast SEO tables as well as the migrations option ( probably best to use the test helper ).
* All our tables should be succesfully created as they were previously.
* Use the `composer generate-migration migration_name` command to create a new migration.
* Write a migration, for example adding a column `test` to the indexables table.
* Run composer install again to make sure your new migration is picked up.
* Delete the yoast migrations free option to ensure migrations are checked again as the version number won't have changed.
* Load any page, the column should be added.
* Create a new file in the migrations directory with the proper filename ( 20200526092459_BadMigration.php ) that simply echo's a message and calls exit. Delete the yoast migrations free option and load a page, you should not see your message.

## UI changes

## Documentation

## Quality assurance


Fixes #15090
Fixes https://github.com/Yoast/wordpress-seo/issues/15273
